app/datatables/neighbourhood_datatable.rb

```diff
@@ -0,0 +1,35 @@
+class NeighbourhoodDatatable < Datatable
+  def view_columns
+    # Declare strings in this format: ModelName.column_name
+    # or in aliased_join_table.column_name format
+    @view_columns ||= {
+      id:              { source: 'Neighbourhood.id', cond: :eq },
+      name:            { source: 'Neighbourhood.name' },
+      unit_name:       { source: 'Neighbourhood.unit_name' },
+      unit_code_key:   { source: 'Neighbourhood.unit_code_key' },
+      unit_code_value: { source: 'Neighbourhood.unit_code_value' },
+    }
+  end
+
+  def data
+    records.map do |record|
+      {
+        id:              link_to(record.id, edit_admin_neighbourhood_path(record)),
+        name:            record.name,
+        unit_name:       record.unit_name,
+        unit_code_key:   record.unit_code_key,
+        unit_code_value: record.unit_code_value,
+        # county:   record.county,
+        # district: record.district,
+        # region:   record.region,
+        # country:  record.country
+      }
+    end
+  end
+
+  def get_raw_records
+    # insert query here
+    # Neighbourhood.all
+    options[:neighbourhoods]
```

Comment:
A little unclear here, but I assume this is scoping records to return by logged in user permmissions?

Reply:
I'm not sure, this is the boilerplate the docs tell you to cut and paste. i think it's just to pass in the page number and per page for the paginator to work.